cookbook/testing/widget-test-introduction.md

```diff
@@ -0,0 +1,266 @@
+---
+layout: page
+title: "Introduction to Widget testing"
+permalink: /cookbook/testing/widget-test-introduction/
+---
+
+In the [introduction to unit testing](/cookbook/testing/unit-test/) recipe, we
+learned how to test Dart classes using the `test` package. In order to test 
+Widget classes, we'll need a few additional tools provided by the 
+[`flutter_test`](https://docs.flutter.io/flutter/flutter_test/flutter_test-library.html)
+package, which ships with the Flutter SDK.
+
+The `flutter_test` package provides the following tools for testing Widgets:
+
+  * The [`WidgetTester`](https://docs.flutter.io/flutter/flutter_test/WidgetTester-class.html),
+  which allows us to build and interact with Widgets in a test environment.
+  * The [`testWidgets`](https://docs.flutter.io/flutter/flutter_test/testWidgets.html) 
+  function. This function will automatically create a new `WidgetTester` for 
+  each test case, and is used in place of the normal `test` function. 
+  * [`Finder`](https://docs.flutter.io/flutter/flutter_test/Finder-class.html)
+  classes. These allow us to search for Widgets in the test environment.  
+  * Widget-specific [`Matcher`](https://docs.flutter.io/flutter/package-matcher_matcher/Matcher-class.html) 
+  constants, which help us verify whether a `Finder` locates a Widget or 
+  multiple Widgets in the test environment.
+  
+If this sounds overwhelming, don't worry! We'll see how all of these pieces fit
+together throughout this recipe.
+  
+### Directions
+
+  1. Add the `flutter_test` dependency
+  2. Create a Widget to test
+  3. Create a `testWidgets` test
+  4. Build the Widget using the `WidgetTester`
+  5. Search for our Widget using a `Finder`
+  6. Verify our Widget is working using a `Matcher`
+  
+### 1. Add the `flutter_test` dependency
+
+Before we can begin writing tests, we'll need to include the `flutter_test` 
+dependency in the `dev_dependencies` section of our `pubspec.yaml` file. If 
+you create a new Flutter project with the command line tools, this dependency
```

Comment:
They do now! In the past, I felt like I'd seen it come and go (maybe I'm wrong there) and had to manually add it. So I wanted to include it in case it was somehow missing. Happy to take it out if you think this is fluff!